torch/testing/_internal/common_methods_invocations.py

```diff
@@ -5085,6 +5098,7 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            dtypes=all_types_and_complex_and(torch.bool, torch.half),
            dtypesIfCUDA=all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16),
            sample_inputs_func=sample_inputs_scatter,
+           supports_complex_autograd=False,
```

Comment:
This will cause a logical merge conflict because this metadata has been generalized here:

https://github.com/pytorch/pytorch/blob/151e81b7bc0ae430c95045c36928e1198b0131be/torch/testing/_internal/common_methods_invocations.py#L161

You want to list all the forward types except the complex types.

I'm confused, however, why does scatter no longer support complex autograd? Or did it never support it and this was improperly tested?

Reply:
I think, for some reason, it never supported... At least, it is not inside the list [`GRADIENT_IMPLEMENTED_FOR_COMPLEX`](https://github.com/pytorch/pytorch/blob/9ec6883442df8ead765941441854a5518e35c69c/tools/autograd/gen_variable_type.py#L81). I guess, I can add it there. Not sure about whether it's correct or not, though.

Reply:
cc @anjali411 to take a look